### ISSUE TITLE
Why neo4j-jdbc-http driver request two times

### ISSUE THREAD

**Issue by eingmarra** [neo4j/neo4j-jdbc#188]
Sorry for my poor English.
I use wireshark and get http request as follow:

POST /db/data/transaction/commit HTTP/1.1
 Accept: application/json; charset=UTF-8
X-Stream: true
Content-Length: 734
Content-Type: application/json; charset=UTF-8
Host: 150.138.123.72:17474
Connection: Keep-Alive
User-Agent: Neo4j JDBC Driver
Accept-Encoding: gzip,deflate

{"statements":[{"statement":"Optional MATCH (u1:User{uid:{1}})    ...     ,"parameters":{"1":205940551,"2":193984160,"3":205940551,"4":193984160,"5":"2019-03-04 17:35:55","6":"2019-03-04 17:35:55","7":"2019-03-04 17:35:55","8":"2019-03-04 17:35:55","9":1551692154382,"10":1551692154382},"includeStats":true}]}

================================================================

POST /db/data/transaction/commit HTTP/1.1
Accept: application/json; charset=UTF-8
X-Stream: true
Content-Length: 734
Content-Type: application/json; charset=UTF-8
Host: 150.138.123.72:17474
Connection: Keep-Alive
User-Agent: Neo4j JDBC Driver
Accept-Encoding: gzip,deflate
_**Authorization: Basic bmVvNGo6OTE4OTgwMGFa**_

{"statements":[{"statement":"Optional MATCH (u1:User{uid:{1}})    ...     ,"parameters":{"1":205940551,"2":193984160,"3":205940551,"4":193984160,"5":"2019-03-04 17:35:55","6":"2019-03-04 17:35:55","7":"2019-03-04 17:35:55","8":"2019-03-04 17:35:55","9":1551692154382,"10":1551692154382},"includeStats":true}]}

only difference is "**Authorization: Basic bmVvNGo6OTE4OTgwMGFa**" ? I am so confused about this. Whats wrong with my java coding? (By the way, i mavened jdbc driver which version is 3.0.0)

Thanks in advance

**Comment by jexp** [comment]
Please share the code you’re using

**Comment by eingmarra** [comment]
i coding a very simple example as follows:

```
public static void main(String... args) throws Exception
{
    	Connection connection = null;
    	PreparedStatement prepareStatement = null;
    	 
	connection = DriverManager.getConnection("jdbc:neo4j:http://150.XXX.XXX.72:7474", "neo4j", "forget it");
	String cypher = "create (user:User{uid:?})";
	try {
    		prepareStatement = connection.prepareStatement(cypher);
        	prepareStatement.setInt(1, 1234);
        } catch (SQLException e) {
			throw new RuntimeException("无法执行graphic.db数据库cypher语句（" + cypher + "）", e);
	}
		
	try {
		prepareStatement.executeUpdate();
	} catch (SQLException e) {
		throw new RuntimeException("无法获取graphic.db数据库cypher语句（" + cypher + "）的结果集", e);
	}finally {
	        if (prepareStatement != null) try { prepareStatement.close(); } catch (SQLException e) {throw new RuntimeException("无法关闭analysis数据库prepareStatement", e);}
	        if (connection != null) try { connection.close(); } catch (SQLException e) {throw new RuntimeException("无法关闭analysis数据库连接", e);}
	}
}
```

**Comment by eingmarra** [comment]
![default](https://user-images.githubusercontent.com/43944755/53850080-0a120000-3ff5-11e9-9245-4816af740f8d.png)

**Comment by eingmarra** [comment]
the **Authorization** header of first request is missing, so it respond "**HTTP1.1 401 Unauthorized**":
![default](https://user-images.githubusercontent.com/43944755/53850185-52c9b900-3ff5-11e9-83c8-ccd26b282d34.png)

**Comment by conker84** [comment]
@eingmarra are you using the version `3.0.0`?

**Comment by eingmarra** [comment]
yes, my jdbc-dirver version is 3.0.0 

> only difference is "Authorization: Basic bmVvNGo6OTE4OTgwMGFa" ? I am so confused about this. Whats wrong with my java coding? (By the way, **i mavened jdbc driver which version is **3.0.0****)

```
<dependencies>
    	...
	<dependency>
		    <groupId>org.neo4j</groupId>
		    <artifactId>neo4j-jdbc-driver</artifactId>
		    <version>3.3.0</version>
	</dependency>
```

**Comment by eingmarra** [comment]
I have tried 3.3.1 and 3.4.0 neo4j-jdbc-driver, but the problem is not solved.

I changed my pom.xml and get the driver from github as follows:
```

	<repositories>
	    <repository>
		    <id>jitpack.io</id>
		    <url>https://jitpack.io</url>
		</repository>
	</repositories>
...
	<dependencies>
		<dependency>
		    <groupId>org.neo4j</groupId>
		    <artifactId>neo4j-jdbc-driver</artifactId>
		    <version>3.4.0</version>
		</dependency>
```

**Comment by conker84** [comment]
Hi @eingmarra and thanks for the catch!
Can you try to use this fix?
https://jitpack.io/#conker84/neo4j-jdbc/issue_188-SNAPSHOT
Looking forward to your feedback!

**Comment by eingmarra** [comment]
I have tried it again, but it seems not be fixed. Hope your next fix version.   :-)

![图片](https://user-images.githubusercontent.com/43944755/54265234-cba9b180-45af-11e9-8da2-22e5f0939ca6.png)
![图片](https://user-images.githubusercontent.com/43944755/54265416-34912980-45b0-11e9-868e-bb7a0130bb9b.png)

**Comment by conker84** [comment]
That's wired looking at the image seems you downloaded the v 3.0.1 which is an old one (not fixed).
Please try with the attached jar.

[neo4j-jdbc-driver-3.4.0.jar.zip](https://github.com/neo4j-contrib/neo4j-jdbc/files/2962947/neo4j-jdbc-driver-3.4.0.jar.zip)

**Comment by eingmarra** [comment]
I've download the v3.4.0-driver which is from the link you give. And maven it as local jar file:
![图片](https://user-images.githubusercontent.com/43944755/54325803-00187e80-463f-11e9-802a-4244232005a8.png)

![图片](https://user-images.githubusercontent.com/43944755/54325682-82547300-463e-11e9-9cfa-e936faccd319.png)

But the problem is still there (**the Authorization header is missing**):
![图片](https://user-images.githubusercontent.com/43944755/54325846-3524d100-463f-11e9-89da-a6a52c8ff8c1.png)

**Comment by conker84** [comment]
Please, can you share your sample project?

**Comment by eingmarra** [comment]
Hi , i am very sorry!
I found I've made a big mistake yesterday. I maven it with my as local jar file, but eclipse seems to not clean absolutely, it still use the neo4j-jdbc-driver from remote site (I found it this morning).

So i recreate a **new** project, and test my code with the jar file you give me. It works !!!!

![图片](https://user-images.githubusercontent.com/43944755/54411564-d177d200-472a-11e9-8490-19275a25de78.png)
![图片](https://user-images.githubusercontent.com/43944755/54411582-da68a380-472a-11e9-99f5-7a283e055f1e.png)

Thank you very much!!!

**Comment by eingmarra** [comment]
By the way, the 3.4.0 driver on https://jitpack.io seems not fix this problem.   
```

<repositories>
	    <repository>
		    <id>jitpack.io</id>
		    <url>https://jitpack.io</url>
		</repository>
	</repositories>

       <dependency>
	       <groupId>com.github.conker84</groupId>
	       <artifactId>neo4j-jdbc</artifactId>
		<version>3.4.0</version>
       </dependency>
```

**Comment by conker84** [comment]
Closed via a4b6b47
